### PR TITLE
Add istioctl profile list test

### PR DESCRIPTION
**Pull request by Monkeyanator**
Adds test for `istioctl profile list` command. Also changes `fmt.Print` to `cmd.Print` in various places to allow for better testing (`fmt.Print` will not capture command output with `SetOut`).

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[x] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

File: operator/cmd/mesh/profile-list_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	var out bytes.Buffer
+	rootCmd.SetOut(&out)
+	rootCmd.SetErr(&out)
+
+	err := rootCmd.Execute()
+	if err != nil {
+		t.Fatalf("failed to execute istioctl profile command: %v", err)
+	}
+	output := out.String()
+	expectedProfiles := []string{"default", "demo", "empty", "minimal", "openshift", "remote"}
```
Comment (shamsher31) [inline operator/cmd/mesh/profile-list_test.go:41]: The current profile list also consists of a `preview` profile.
https://preliminary.istio.io/latest/docs/setup/additional-setup/config-profiles/

```suggestion
	expectedProfiles := []string{"default", "demo", "empty", "minimal", "openshift", "preview", "remote"}
```

Comment (Monkeyanator) [reply operator/cmd/mesh/profile-list_test.go]: Thanks for catching, fixed

**Review by shamsher31 — APPROVED** [review]

**Comment by shamsher31** [comment]
@Monkeyanator you are still working on it or you want to remove `hold`?

**Pull request merged**